pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1221,6 +1211,23 @@ func Predicates(pod *v1.Pod, nodeInfo *schedulercache.NodeInfo) (bool, []algorit
 	return len(predicateFails) == 0, predicateFails, nil
 }
 
+func NodeConditionPredicates(nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason) {
+	reasons := []algorithm.PredicateFailureReason{}
+
+	for _, c := range nodeInfo.Node().Status.Conditions {
+		// TODO: There are other node status that the DaemonSet should ideally respect too,
+		//       e.g. MemoryPressure, and DiskPressure
+		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
```

Comment:
slack with gmarek@, we will add 'MemoryPressure' and 'DiskPressure' toleration in admission control for best effort pods in #42001 .